Sources/iron/math/Quat.hx

```diff
@@ -255,6 +257,150 @@ class Quat {
 		return this;
 	}
 
+
+	/**
+	  Convert this quaternion to an Euler of arbitrary order.
+	  @param	the order of the euler to obtain
+			(in blender order, opposite from mathematical order)
+			can be "XYZ", "XZY", "YXZ", "YZX", "ZXY", or "ZYX".
+	  @return	A new YZX Euler that represents the same rotation as this
+				quaternion.
+	**/
+	// this method use matrices as a middle ground
+	// (and is copied from blender's internal code in mathutils)
+	// note: there are two possible eulers for the same rotation, blender defines the 'best' as the one with the smallest sum of absolute components
+	//	should we actually make that choice, or is just getting one of them randomly good?
+	// note2: it seems that this engine transforms a vector by using vector×matrix instead of matrix×vector, meaning that the outer transformations are on the RIGHT.
+	//	(…Except for quaternions, where the outer quaternions are on the LEFT.)
+	//	anywho, the way the elements of the matrix are ordered makes sense (first digit-> row ID, second digit->column ID) in this system.
+	public inline function toEulerOrdered(p:String): Vec4{
```

Comment:
Very cool!

In theory, the string parameter could be replaced with an `enum abstract` over int for slightly better performance and a bit more type safety, two bits could encode one axis each and then you can use bit masks to get the value at each position. Combined with a switch case instead of if/else ifs this could be even faster.

Enum properties in Blender can also have explicit int values (they are stored as int's internally), however I don't know if the value can be received that easily without a custom conversion function. But even that could be better as then the conversion from string to int then happens during export-time.

It's just an idea, I don't know all the details right now :)

Reply:
er… looking at this, but I can't find a way of doing this without a janky special case in `make_logic.py` (or similar) just yet.
maybe this could be fixed when handling of the properties will be a little smarter?
like, stop having `node_utils.get_haxe_property_names` use `range(0,10)` and such.

Reply:
Yeah that's what I feared. It's probably better anyway to do this for all enums in the future, at least if there is a performance difference compared to using strings.